examples/auth/oauth2/client-secret-ref/secret.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: oauth2-secrets
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: auth-oauth2-oauth0-client-secret-ref-api
+  namespace: oauth2-secrets
+type: Opaque
+data:
+  client_secret: "WjZNWDdOcmVKdW1XTG1mNnVuc1E1dWlFVXJUQnhmTnRxRzlWeTVLamt0bnZmai1fZlJDQk85RVUxbUwxWXpBSgo="
```

Comment:
`base64 -w0` version of `Z6MX7NreJumWLmf6unsQ5uiEUrTBxfNtqG9Vy5Kjktnvfj-_fRCBO9EU1mL1YzAJ`.

Once we retrieve the secret Kubernetes will return it un-decoded form.

Reply:
Is this totally necessary? No other way of just saving the plain text?

Reply:
Yes it is.

You need to `base64` encode the data. The key then to retrieve the data from the `Secret` is `data[0]` as a `string`. Kubernetes will then return this <https://github.com/kubeshop/kusk-gateway/pull/840/files#diff-2bc3860e7e2c113912b1e720875b3185351a85f9bee3ae97d97be1506979aa83R32> (the original `client_secret`).